packages/color/src/Color.ts

```diff
@@ -0,0 +1,331 @@
+    get green(): number
+    {
+        return this._components[1];
+    }
+
+    /** Read out blue value (0 - 1) */
+    get blue(): number
+    {
+        return this._components[2];
+    }
+
+    /** Read out alpha value (0 - 1) */
+    get alpha(): number
+    {
+        return this._components[3];
+    }
+
+    /**
+     * Set the value, suitable for chaining
+     * @param value
+     */
+    setValue(value: ColorSource): this
+    {
+        this.value = value;
+
+        return this;
+    }
+
+    /**
+     * Set the current color source
+     * @type {PIXI.ColorSource}
+     */
+    set value(value: ColorSource)
+    {
+        // Support copying from other Color objects
+        if (value instanceof Color)
+        {
+            this._value = value._value;
+            this._int = value._int;
+            this._components.set(value._components);
+        }
+        else if (this._value !== value)
+        {
+            this.normalize(value);
+            this._value = value;
+        }
+    }
+    get value(): Exclude<ColorSource, Color>
+    {
+        return this._value;
+    }
+
+    /**
+     * Get RGBA color object.
+     * @example
+     * import { Color } from 'pixi.js';
+     * new Color('white').toRgb(); // returns { r: 1, g: 1, b: 1, a: 1 }
+     */
+    toRgba(): RgbaColor
+    {
+        const [r, g, b, a] = this._components;
+
+        return { r, g, b, a };
+    }
+
+    /**
+     * Get RGB color object.
+     * @example
+     * import { Color } from 'pixi.js';
+     * new Color('white').toRgb(); // returns { r: 1, g: 1, b: 1 }
+     */
+    toRgb(): RgbColor
+    {
+        const [r, g, b] = this._components;
+
+        return { r, g, b };
+    }
+
+    /** Get the color as a CSS-style rgba string: `rgba(255,255,255,1.0)`. */
+    toRgbaString(): string
+    {
+        const [r, g, b, a] = this._components;
+
+        return `rgba(${Math.round(r * 255)},${Math.round(g * 255)},${Math.round(b * 255)},${a})`;
+    }
+
+    /**
+     * Converts color to an [R, G, B] array of normalized floats (numbers from 0.0 to 1.0).
+     * @example
+     * import { Color } from 'pixi.js';
+     * new Color('white').toRgbArray(); // returns [1, 1, 1]
+     * @param {number[]|Float32Array} [out] - Output array
+     */
+    toRgbArray<T extends (number[] | Float32Array) = number[]>(out?: T): T
+    {
+        out = out ?? [] as T;
+        const [r, g, b] = this._components;
+
+        out[0] = r;
+        out[1] = g;
+        out[2] = b;
+
+        return out;
+    }
+
+    /**
+     * Converts a color to a hexadecimal number.
+     * @example
+     * import { Color } from 'pixi.js';
+     * new Color('white').toNumber(); // returns 16777215
+     */
+    toNumber(): number
+    {
+        return this._int;
+    }
+
+    /**
+     * Converts a hexadecimal color number to a string.
+     * @example
+     * import { Color } from 'pixi.js';
+     * new Color('white').toString(); // returns "#ffffff"
+     */
+    toString(): string
```

Comment:
I would expect implicit string conversion to be lossless (i.e. not losing the alpha channel):

```
expect(`${new Color('#ffffff00')`).toEqual('#ffffff00');
```

I suggest to rename this method to `toHexString()`, rename the `toHexString()` to `toHexAlphaString()` and make the `toString()` method an alias for the `toHexAlphaString()` method.

Or `toRgbHexString()`, `toRgbaHexString()` and make `toString()` an alias for the `toRgbaHexString()`.

WDYT ?

Reply:
I don't have strong feelings about these names and would like feedback from others. I was thinking `toString()` as equivalent to `toNumber()`(non-alpha).

Reply:
I'd go with something like:

```js
toHex(); // "#ff0000"
toHexa(); // "#ff0000ff"
toRgb();
toRgba();
toNumber();
```
imo `toString()` should expose all properties including alpha, the main purpose is presentation.

```
toString(); // "Color(r: 255, g: 0, b: 0, a: 1)"
```
I don't like assuming preference/usage, e.g. `red` is also a string just like `#ff0000` or `rgb(255,0,0)`.

---

Also I'd use range `0-255` by default and expose a `normalize()` method to change range to `0-1`.

Reply:
Welcome back @vikpe, thanks for. your thoughts 👍

Reply:
The reason I used floats internally instead of uints was because most of our uses-cases for color tend to utilize floats (uniforms, shaders, etc). I was trying to avoid having to call `normalize`, which makes for a much more verbose API.

I agree with the toString(), that shouldn't mess with standard JS presentation.

Reply:
Then floats it is, looking good!